clients/fides-js/src/lib/events.ts

```diff
@@ -27,14 +27,21 @@ declare global {
   }
 }
 
+/**
+ * Defines the type of "extra" details that can be optionally added to certain
+ * events. This is intentionally vague, but constrained to be basic (primitive)
+ * values for simplicity.
+ */
+export type FidesEventExtraDetails = Record<string, string | number | boolean>;
+
 /**
  * Defines the properties available on event.detail. Currently the FidesCookie
  * and an extra field `meta` for any other details that the event wants to pass
  * around.
  */
 export type FidesEventDetail = FidesCookie & {
   debug?: boolean;
-  extraDetails?: Record<string, string | boolean>;
+  extraDetails?: FidesEventExtraDetails;
```

Comment:
FYI @allisonking: this is just a bit of extra type safety for ourselves. No meaningful change here.